### PR TITLE
Update docker for default admin creds update

### PR DESCRIPTION
**Pull request by derek-ho**
# Changes

Passes in an initial admin password via env variable, implemented in security plugin main and 2.x lines, for 2.12.0 release.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [ ] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Comment by derek-ho** [comment]
@YANG-DB can you help me understand what other changes this repo needs? It doesn't seem to follow the branching strategy of main/2.x - so do we need to handle logic to determine whether to use the default admin password, or the env variable password in the ES_PASSWORD field? Any other places you see that would need changes? Thanks!

**Comment by derek-ho** [comment]
https://github.com/opensearch-project/security/issues/3624

**Comment by YANG-DB** [comment]
@derek-ho please use the `.env` for all the password’s place holders

**Comment by DarshitChanpura** [comment]
> @derek-ho please use the .env for all the password’s place holders

@derek-ho Could you please address this comment.

**Comment by derek-ho** [comment]
Marking this as draft until variable OPENSEARCH_VERSION changes to >= 2.12.0.

**Comment by derek-ho** [comment]
Done. @YANG-DB can you take another look? I also changed the image from staging to the actual released docker images, similar to the dashboards image (which is not using staging).

**Review by YANG-DB — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -723,7 +723,7 @@ services:
 
   # OpenSearch store - engine
   opensearch:
-    image: opensearchstaging/opensearch:${OPENSEARCH_VERSION}
```
[the remark this answers is not in the document]
Comment (derek-ho) [reply docker-compose.yml:726]: Yes, it is a mismatch between OS backend and OSD image

**Comment by derek-ho** [comment]
@YANG-DB can you double check again and we can merge if everything looks good?

**Comment by DarshitChanpura** [comment]
@YANG-DB @pjfitzgibbons @ps48 @kavithacm @joshuali925 @dai-chen @rupal-bq @mengweieric @vamsi-amazon @Swiddis Could we get this reviewed as we are targeting 2.12 for this change?

**Comment by DarshitChanpura** [comment]
@YANG-DB @pjfitzgibbons @ps48 @kavithacm @joshuali925 @dai-chen @rupal-bq @mengweieric @vamsi-amazon @Swiddis Still waiting for reviews. Could we get this reviewed as we are targeting 2.12 for this change?

**Pull request merged**